### PR TITLE
feat: fully qualified name util

### PR DESCRIPTION
**Pull request by sirtorry**
Fixes b/170954330.

### CODE REVIEW

**Comment by sirtorry** [comment]
hmm i ran black again but nothing changed 🤔

**Review by ivanmkc — APPROVED** [review]
Thanks for cleaning up the error messages, looking good!

LGTM

File: google/cloud/aiplatform/utils.py [hunk 1]
```diff
@@ -97,6 +98,62 @@ def extract_fields_from_resource_name(
     return fields
 
 
+def full_name(
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/utils.py:101]: nit: full_resource_name

File: google/cloud/aiplatform/utils.py [hunk 2]
```diff
@@ -97,6 +98,62 @@ def extract_fields_from_resource_name(
...
+    resource_noun: Optional[str] = None,
+    project: Optional[str] = None,
+    location: Optional[str] = None,
+) -> str:
+    """
+    Returns fully qualified resource name.
+
+    Args:
+        resource_name (str):
+            Required. A partial or fully-qualified AI Platform (Unified)
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/utils.py:112]: nit: AI Platform Resource id or fully-qualified resource name.

File: google/cloud/aiplatform/utils.py [hunk 3]
```diff
@@ -97,6 +98,62 @@ def extract_fields_from_resource_name(
...
+            Optional project to retrieve resource_noun from. If not set, project
+            set in aiplatform.init will be used.
+        location (str):
+            Optional location to retrieve resource_noun from. If not set, location
+            set in aiplatform.init will be used.
+
+    Returns:
+        resource_name (str):
+            A fully-qualified AI Platform (Unified) resource name.
+    """
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/utils.py:128]: Raises section.

File: google/cloud/aiplatform/datasets.py [hunk 4]
```diff
@@ -65,28 +64,13 @@ def __init__(
                 credentials set in aiplatform.init.
         """
 
-        # Fully qualified dataset name, i.e. "projects/.../locations/.../datasets/12345"
-        valid_name = extract_fields_from_resource_name(
-            resource_name=dataset_name, resource_noun="datasets"
+        dataset_name = full_name(
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/datasets.py:67]: Please add this to the same logic to the Model class. Unless you plan to add it to the open Endpoint PR.

File: google/cloud/aiplatform/utils.py [hunk 5]
```diff
@@ -97,6 +98,62 @@ def extract_fields_from_resource_name(
...
+        )
+    # project not specified
+    elif not project:
+        raise ValueError("Please provide a project ID")
+    # Invalid resource_noun parameter
+    elif not valid_name:
+        error_message = "Please provide a valid {resource_noun} name or ID".format(
+            resource_noun=resource_noun[:-1],
+        )
+        raise ValueError(error_message)
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/utils.py:152]: Do we also need to check for a non-None resource_noun? Or possibly require it?

File: tests/unit/aiplatform/test_utils.py [hunk 6]
```diff
@@ -112,3 +112,47 @@ def test_invalid_region_raises_with_invalid_region():
 
 def test_invalid_region_does_not_raise_with_valid_region():
     aiplatform.utils.validate_region(region="us-central1")
+
```
Comment (sasha-gitg) [inline tests/unit/aiplatform/test_utils.py:115]: Test for failure modes that test the ValueError exceptions.

**Review by sasha-gitg — CHANGES_REQUESTED** [review]

File: google/cloud/aiplatform/datasets.py (hunk 4 above)
Comment (sirtorry) [reply google/cloud/aiplatform/datasets.py:67]: will put it in the endpoint pr

**Review by vinnysenthil — CHANGES_REQUESTED** [review]

**Review by sasha-gitg — APPROVED** [review]

**Review by vinnysenthil — APPROVED** [review]

**Pull request merged**